Added documentation explaining the requirements to make the kubeconfig generated by Okteto with the new credential plugin within the CLI (#474)
okteto/docs: src/content/cloud/credentials.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> This section explains how to download your Kubernetes credentials and start developing your applications in Okteto  with your favorite CLI tools.

Reviewer: ```suggestion
This section explains how to download your Kubernetes credentials and start developing your applications in Okteto with your favorite CLI tools.
```

> In order to do so, the first thing you need is to install Okteto CLI and ensure the executable is in your PATH. This is needed because the kubeconfig provided by Okteto uses a [credential plugin](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) to authenticate access your Kubernetes cluster. It is a common pattern used by many Kubernetes providers, such as Google Kubernetes Engine (GKE), Azure Kubernetes Service (AKS), or Amazon Elastic Kubernetes Service (EKS) to connect to Kubernetes clusters.

Reviewer: ```suggestion
The first thing you need to do is install the Okteto CLI and ensure the executable is in your PATH. This is needed because the kubeconfig provided by Okteto uses a [credential plugin](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) to authenticate access your Kubernetes cluster. It is a common pattern used by many Kubernetes providers, such as Google Kubernetes Engine (GKE), Azure Kubernetes Service (AKS), or Amazon Elastic Kubernetes Service (EKS) to connect to Kubernetes clusters.
```

## Using your Kubernetes credentials

> Before using your Kubernetes credentials, it is important to know that the kubeconfig file generated by Okteto uses a [credential plugin](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) to authenticate you via Okteto to access your Kubernetes cluster. It is a common pattern used by many Kubernetes providers, such as Google Kubernetes Engine (GKE), Azure Kubernetes Service (AKS), or Amazon Elastic Kubernetes Service (EKS) to connect to Kubernetes clusters in a more secure way. For this reason, it is important that the Okteto CLI is in your PATH and connected to your Okteto instance (having executed `okteto context use ...`).

Reviewer 2: I think that the requirement of having the CLI on the path is a bit hidden. How about we structure it something like this?

```
# Download the CLI
.....

# Add the Okteto CLI to your PATH
(instructions and lightweight explanation)

# Connect to your Okteto instance
okteto context use ....

# Get your Kubeconfig file
okteto kubeconfig ....

Author: Yeah, makes sense. I've changed a bit the structure to follow your recommendation. Let me know what you think @rberrelleza

Once you've downloaded your credentials and it is configured, you'll have full access to your Kubernetes namespaces on Okteto directly from your terminal, using the tools you're already familiar with, such as `helm`, `kubectl`, `okteto` or `skaffold`.

For example, you could use `kubectl` to deploy our `hello-world` application:

```terminal
kubectl apply -f https://raw.githubusercontent.com/okteto/go-getting-started/master/k8s.yml
```

> > As mentioned before, credential plugins are a way of managing Kubernetes cluster credentials in a more secure way as you get time-bound tokens for your service account instead of using auto-generated secret-based ones. In case you face any issue with Okteto's credential plugin or you just simply want to keep using static auto-generated tokens, you can do it setting the environment variable `OKTETO_USE_STATIC_KUBETOKEN` to `true` before running any Okteto command. Be aware that using those static tokens are not recommended by Kubernetes and you will start getting warnings in your `kubectl` output from version `1.27` encouraging you to stop using them.

Reviewer 2: nit: can we include a link to the k8s docs on this?

Author: Yes, I can add it. It was already added in the other paragraph but I'll include it also here

> > As mentioned before, credential plugins are a way of managing Kubernetes cluster credentials in a more secure way as you get time-bound tokens for your service account instead of using auto-generated secret-based ones. In case you face any issue with Okteto's credential plugin or you just simply want to keep using static auto-generated tokens, you can do it setting the environment variable `OKTETO_USE_STATIC_KUBETOKEN` to `true` before running any Okteto command. Be aware that using those static tokens are not recommended by Kubernetes and you will start getting warnings in your `kubectl` output from version `1.27` encouraging you to stop using them.

Reviewer 2: ```suggestion
> As mentioned before, credential plugins are a way of managing Kubernetes cluster credentials in a more secure way as you get time-bound tokens for your service account instead of using auto-generated secret-based ones. In case you face any issue with Okteto's credential plugin or you just want to keep using static auto-generated tokens, you can do it by setting the environment variable `OKTETO_USE_STATIC_KUBETOKEN` to `true` before running any Okteto command. Be aware that using those static tokens is not recommended by Kubernetes, and you will start getting warnings in your `kubectl` output from version `1.27` encouraging you to stop using them.
```

> ## Using your Kubernetes credentials

Reviewer 3: should we add info about use `OKTETO_USE_STATIC_KUBETOKEN` flag?

Reviewer 3: `OKTETO_USE_STATIC_KUBETOKEN` is a temporal feature flag that allows the user to use the old auth method using static tokens

Author: Already added documentation for the flag. I'll resolve the comment

kubectl apply -f https://raw.githubusercontent.com/okteto/go-getting-started/master/k8s.yml
```

> > As mentioned before, [credential plugins](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) are a way of managing Kubernetes cluster credentials in a more secure way as you get time-bound tokens for your service account instead of using auto-generated secret-based ones. In case you face any issue with Okteto's credential plugin, or you just want to keep using static auto-generated tokens, you can do it by setting the environment variable `OKTETO_USE_STATIC_KUBETOKEN` to `true` before running any Okteto command. Be aware that using those static tokens is not recommended by Kubernetes, and you will start getting warnings in your `kubectl` output from version `1.27` encouraging you to stop using them.

Reviewer: ```suggestion
[Credential plugins](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) are a way of managing Kubernetes cluster credentials in a more secure way as you get time-bound tokens for your service account instead of using auto-generated secret-based ones. If you prefer to use static, auto-generated tokens you can do so by setting the environment variable `OKTETO_USE_STATIC_KUBETOKEN` to `true` before running any Okteto command. Be aware that using static tokens is not recommended by Kubernetes and you will receive warnings in your `kubectl` output from version `1.27` onward encouraging you to stop using them.
```
